Mjml.Net/Components/Body/SectionComponent.cs

```diff
@@ -1,11 +1,595 @@
+        public string TextPadding = "4px 4px 4px 0";
+
+        [Bind("css-class")]
+        public string? CssClass;
+
         public override void Render(IHtmlRenderer renderer, GlobalContext context)
         {
+            if (IsFullWidth())
+            {
+                RenderFullWidth(renderer, ref context);
+            }
+            else
+            {
+                RenderSimple(renderer, ref context);
+            }
+        }
+
+        private void RenderFullWidth(IHtmlRenderer renderer, ref GlobalContext context)
+        {
+            var hasBackground = HasBackground();
+
+            var tableElement = renderer.ElementStart("table")
+                .Attr("align", "center")
+                .Attr("background", BackgroundUrl)
+                .Attr("border", "0")
+                .Attr("cellpadding", "0")
+                .Attr("cellspacing", "0")
+                .Attr("role", "presentation")
+                .Class(CssClass)
+                .Style("width", "100%")
+                .Style("border-radius", BorderRadius);
+
+            if (hasBackground)
+            {
+                tableElement
+                    .Style("background", GetBackground())
+                    .Style("background-color", BackgroundColor)
+                    .Style("background-position", BackgroundPosition)
+                    .Style("background-repeat", BackgroundRepeat)
+                    .Style("background-size", BackgroundSize);
+            }
+            else
+            {
+                tableElement
+                    .Style("background", BackgroundColor)
+                    .Style("background-color", BackgroundColor);
+            }
+
+            renderer.ElementStart("tbody");
+            renderer.ElementStart("tr");
+            renderer.ElementStart("td");
+
+            if (hasBackground)
+            {
+                RenderSectionWithBackground(renderer, ref context);
+            }
+            else
+            {
+                RenderSectionStart(renderer, ref context);
+                RenderSection(renderer, ref context);
+                RenderSectionEnd(renderer);
+            }
+
+            renderer.ElementEnd("td");
+            renderer.ElementEnd("tr");
+            renderer.ElementEnd("tbody");
+            renderer.ElementEnd("table");
+        }
+
+        private void RenderSimple(IHtmlRenderer renderer, ref GlobalContext context)
+        {
+            RenderSectionStart(renderer, ref context);
+
+            if (HasBackground())
+            {
+                RenderSectionWithBackground(renderer, ref context);
+            }
+            else
+            {
+                RenderSection(renderer, ref context);
+            }
+
+            RenderSectionEnd(renderer);
+        }
+
+        private void RenderSectionStart(IHtmlRenderer renderer, ref GlobalContext context)
+        {
+            var containerWidth = context.GetContainerWidth();
```

Comment:
The container width is needed so often. Therefore I just put in a variable. See GroupComponent.cs